pkg/vsphere/disk/disk_manager.go

```diff
@@ -223,6 +223,20 @@ func (m *Manager) Create(ctx context.Context, newDiskURI string,
 //	return nil
 // }
 
+func (m *Manager) Attach(ctx context.Context, spec *types.VirtualDisk) error {
+	machineSpec := configureDeviceSpec(ctx, *m.vm, types.VirtualDeviceConfigSpecOperationAdd, types.VirtualDeviceConfigSpecFileOperationCreate, spec)
+
+	_, err := tasks.WaitAndRetryForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
```

Comment:
The storage changes are actually just sections of code that are somewhat extracted from govmomi since the reconfigure and the spec creation needed to be done separately in ordered to perform the WaitAndRetry on the actual Reconfigure. We needed the exposed Task object which was hidden in the old call path.